packages/roosterjs-editor-api/lib/format/setFontName.ts

```diff
@@ -8,11 +9,18 @@ import { IEditor } from 'roosterjs-editor-types';
  * Currently there's no validation to the string, if the passed string is invalid, it won't take affect
  */
 export default function setFontName(editor: IEditor, fontName: string) {
+    const parentNodes: Node[] = [];
     fontName = fontName.trim();
     // The browser provided execCommand creates a HTML <font> tag with face attribute. <font> is not HTML5 standard
     // (http://www.w3schools.com/tags/tag_font.asp). Use applyInlineStyle which gives flexibility on applying inline style
     // for here, we use CSS font-family style
     applyInlineStyle(editor, (element, isInnerNode) => {
         element.style.fontFamily = isInnerNode ? '' : fontName;
+
+        if (parentNodes.indexOf(element.parentNode) === -1) {

```

Comment:
This still can't sole the problem when LI is not the direct parent. e.g.
```html
<ol>
<li><div><span>aaa</span></div>
</li>
<ol>
```

Reply:
Fixed this adding a new function to get the closest element ancestor of a specific tag.

Reply:
Also added a test case for this specific scenario

Reply:
Now this check is not right. You need to check parentListItem instead of element.parentNode